src/jit/compiler.h

```diff
@@ -3510,7 +3510,7 @@ class Compiler
 #ifdef _TARGET_ARM64_
     InstructionSet lookupHWIntrinsicISA(const char* className);
     NamedIntrinsic lookupHWIntrinsic(const char* className, const char* methodName);
-    bool impCheckImmediate(GenTree* immediateOp, unsigned int max);
+    GenTree* addRangeCheckIfNeeded(GenTree* lastOp, unsigned int max, bool mustExpand);
```

Comment:
Once the Arm64 intrinsics are also table-driven (I *think* there's an issue for that?) it would make sense to just declare and implement this once.

Reply:
It is not quite the same signature as in the x64 version (didn't need to be quite as flexible). They could converge, I think.